botocore/exceptions.py

```diff
@@ -342,17 +342,30 @@ class UnsupportedSignatureVersionError(BotoCoreError):
 class ClientError(Exception):
     MSG_TEMPLATE = (
         'An error occurred ({error_code}) when calling the {operation_name} '
-        'operation: {error_message}')
+        'operation{retry_info}: {error_message}')
```

Comment:
Yeah.  The first version I had rendered the msg, but before calling `super()` would mess with the msg to inject retry info.  This approach seemed less worse that the first approach.

Also open to other ways to improve this.